src/controllers/handlePaymentsController.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express';
+import Stripe from 'stripe';
+import { config } from 'dotenv';
+import { Product } from '../database/models/Product';
+import { Size } from '../database/models/Size';
+import Order from '../database/models/order';
+import { sendInternalErrorResponse } from '../validations';
+
+config();
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY as string);
+
+type Items = {
+  id: string;
+  quantity: number;
+}[];
+
+const fetchProducts = async (items: Items) => {
+  const products = await Promise.all(
+    items.map(async item => {
+      return await Product.findAll({
+        where: { id: item.id },
+        include: {
+          model: Size,
+          as: 'sizes',
+          attributes: ['price', 'quantity', 'id'],
+        },
+      });
+    })
+  );
+  return products.flat();
+};
+
+const updateStockLevels = async (items: Items) => {
+  const products = await fetchProducts(items);
+  for (const item of items) {
+    const product = products.find(p => p.id === item.id);
+    if (product) {
+      const newQuantity = product.sizes[0].quantity - item.quantity;
```

Comment:
selecting the very first size I think, won't be the right approach to getting the the quantity affected, try incorporating the size id, so you can be accurate on the real size affected